src/components/Layout/Header.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'gatsby';
+import logoBlack from '@static/logo-black.png';
+import logoSmall from '@static/logo-small.png';
+
+// TODO: this declaration should live over a d.ts file to prevent cluttering
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace JSX {
+    interface IntrinsicElements {
+      'dc-header': unknown;
+      'dc-footer': unknown;
+    }
+  }
+}
+
+const GATSBY_HOST_URL = process.env.GATSBY_HOST_URL;
+const GATSBY_COMMUNITY_URL = process.env.GATSBY_COMMUNITY_URL;
+
+const GUEST_LINKS = [
+  {
+    href: '/debt-union',
+    text: 'Join the Union',
+    target: '_self'
+  }
+];
+
+const USER_LINKS = [
+  {
+    href: '/hub',
+    text: 'Member hub',
+    target: '_self'
+  }
+];
+
+const HEADER_LINKS = [
+  {
+    href: 'https://biden100.debtcollective.org/',
+    text: 'Biden Jubilee 100',
+    target: '_blank'
+  },
+  {
+    href: 'https://community.debtcollective.org/',
+    text: 'Community',
+    target: '_blank'
+  },
+  {
+    href: 'https://teespring.com/stores/debt-collective',
+    text: 'Store',
+    target: '_blank'
+  }
+];
+
+const Header = () => {
+  const [links, setLinks] = useState(HEADER_LINKS);
+  const [user, setUser] = useState(null);
+
+  const handleUserSynced = (e: unknown) => {
+    // @ts-ignore
+    const syncedUser = e?.detail;
+    setUser(syncedUser);
+  };
+
+  // TODO: https://stenciljs.com/docs/events JSX event binding not working
+  useEffect(() => {
+    const header = document.querySelector('#dc-header');
+    header?.addEventListener('userSynced', handleUserSynced);
+
+    return () => {
+      window.removeEventListener('userSynced', handleUserSynced);
+    };
+  }, []);
+
+  useEffect(() => {
+    const extraLinks = user ? USER_LINKS : GUEST_LINKS;
```

Comment:
Maybe I need to keep `membership?.id` check here.
```
const extraLinks = (membership?.id || user.id) ? USER_LINKS : GUEST_LINKS;
```

Reply:
mmm this shouldn't be the case anymore, a User should always have a Discourse account now.

This `membership.id` comes from the `/users/current` endpoint on the membership app right?

Reply:
Then I will bet it is only my local and we can go with the User's discourse account.
And yes, the `membership.id` comes from the endpoint you describe on the membership app.

_Should I merge then?_

Reply:
maybe we can have a quick call to review this? I think we can remove the functionality of making a request to the membership app to request for session

Reply:
I am not requesting to membership to check for session. 🤔